insim/src/insim/ipb.rs

```diff
@@ -0,0 +1,120 @@
+use std::{default::Default, net::Ipv4Addr};
+
+use indexmap::{set::Iter as IndexSetIter, IndexSet};
+use insim_core::binrw::{self, binrw, BinRead, BinResult, BinWrite};
+
+use crate::identifiers::RequestId;
+
+const IPB_MAX_BANS: usize = 120;
+
+#[binrw::parser(reader, endian)]
+fn binrw_parse_ipb_bans(count: u8) -> BinResult<IndexSet<Ipv4Addr>> {
+    let mut data = IndexSet::new();
+    for _i in 0..count {
+        let ip = Ipv4Addr::from(u32::read_options(reader, endian, ())?);
+        let _ = data.insert(ip);
+    }
+    Ok(data)
+}
+
+#[binrw::writer(writer, endian)]
+fn binrw_write_ipb_bans(input: &IndexSet<Ipv4Addr>) -> BinResult<()> {
+    for i in input.iter() {
+        u32::from(*i).write_options(writer, endian, ())?;
+    }
+
+    Ok(())
+}
+
+#[binrw]
+#[bw(assert(banips.len() <= IPB_MAX_BANS))]
+#[derive(Debug, Clone, Default, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
+/// Mods Allowed - restrict the mods that can be used
+pub struct Ipb {
+    /// Non-zero if the packet is a packet request or a reply to a request
+    pub reqi: RequestId,
+
+    /// Number of bans in this packet, from the wire
+    /// This value is not to be trusted as we use an IndexSet to record the bans internally. It is technically
+    /// possible that LFS could return duplicate entries, but we have no way of verifying that.
+    #[bw(calc = banips.len() as u8)]
+    #[brw(pad_after = 4)]
+    numb: u8,
+
+    #[br(parse_with = binrw_parse_ipb_bans, args(numb))]
+    #[bw(write_with = binrw_write_ipb_bans)]
+    banips: IndexSet<Ipv4Addr>,
+}
+
+impl Ipb {
+    /// Returns `true` if a Vehicle is contained in this packet
```

Comment:
The comments in this packet definition still refer to "vehicle" and "mods", that looks like an oversight after a copy&paste.